src/box/memtx_space.c

```diff
@@ -1171,13 +1198,37 @@ memtx_build_on_replace_rollback(struct trigger *trigger, void *event)
 	return 0;
 }
 
+/**
+ * Deletes the txn triggers from memtx_ddl_state::txn_triggers
+ * since they are going to be deleted.
+ */
+static int
+memtx_build_on_replace_commit(struct trigger *trigger, void *event)
+{
+	(void)event;
+	struct index_build_txn_trigger_data *data = trigger->data;
+	rlist_del(data->in_state);
+	data->in_state = NULL;
+	return 0;
+}
+
 /*
  * Struct to allocate by memtx_build_on_replace trigger
- * (on_rollback trigger with its data).
+ * (transactional triggers with their data).
  */
-struct on_rollback_trigger_with_data {
+struct txn_triggers_with_data {
```

Comment:
I think `txn_trigger_with_data` and `index_build_txn_trigger_data` should be merged. You can access the structure form the trigger with `container_of` and `txn_stmt` is passed in `event`.